src/battle_game/core/player.h

```diff
@@ -1,9 +1,31 @@
 #pragma once
 #include "cstdint"
 #include "input_data.h"
+#include "map"
+#include "vector"
 
 namespace battle_game {
 class GameCore;
+class Buff {
+ public:
+  Buff(const char *BuffName,
+       float Time,
+       bool IsSupportManyLayer,
+       bool IsInfiniteTime,
+       int RemoveWhenDie)
+      : BuffName_(BuffName),
+        Time_(Time),
+        IsSupportManyLayer_(IsSupportManyLayer),
+        IsInfiniteTime_(IsInfiniteTime),
+        RemoveWhenDie_(RemoveWhenDie) {
+  }
+
+  const char *BuffName_;
+  float Time_;
+  bool IsSupportManyLayer_;
+  bool IsInfiniteTime_;
+  int RemoveWhenDie_;
+};
```

Comment:
put these codes into new files: buff.h and buff.cpp
make variables private
private variable names should be in lower_case_ endding with an underscore.